src/datatrove/utils/binaryio.py

```diff
@@ -29,6 +30,14 @@ def read_tuples_from_file(file: BinaryIO, *formats, lines_to_buffer: int = 5):
         yield from reader.iter_unpack(chunk)
 
 
+def read_np_from_file(file: AbstractBufferedFile, dtype, lines_to_buffer: int = 5):
```

Comment:
maybe reuse this version: https://github.com/huggingface/datatrove/blob/main/src/datatrove/pipeline/dedup/sentence_dedup.py#L304